changelog/assert_ctfe.dd

```diff
@@ -0,0 +1,20 @@
+ctfe-only function elision
```

Comment:
```suggestion
Added ability to specify that a function is CTFE-only function
```

Maybe it's just me but I found the previous sentence slightly ambiguous, and a bit too ad-hoc for a user-facing changelog.

Reply:
I've completely rewritten the changelog, I hope this addresses this problem.